server/data/data-jmap/src/main/java/org/apache/james/jmap/api/filtering/Rule.java

```diff
@@ -19,49 +19,318 @@
+            
+            public static Optional<Comparator> find(String comparatorName) {
+                return Arrays.stream(values())
+                        .filter(value -> value.comparatorName.equalsIgnoreCase(comparatorName))
+                        .findAny();
+            }
+            
+            public static Comparator of(String comparatorName) {
+                return find(comparatorName).orElseThrow(() -> new IllegalStateException("'" + comparatorName + "' is not a valid comparator name"));
+            }
+            
+            private final String comparatorName;
+            
+            private Comparator(String comparator) {
+                this.comparatorName = comparator;
+            }
+            
+            public String asString() {
+                return comparatorName;
+            }
+        }
+
+        public static Condition of(Field field, Comparator comparator, String value) {
+            Preconditions.checkNotNull(field, "field should no be null");
+            Preconditions.checkNotNull(comparator, "comparator should no be null");
+            Preconditions.checkNotNull(value, "value should no be null");
+            Preconditions.checkArgument(StringUtils.isNotBlank(value), "value should no be empty");
+            return new Condition(field, comparator, value);
+        }
+
+        private final Field field;
+        private final Comparator comparator;
+        private final String value;
+
+        private Condition(Field field, Comparator comparator, String value) {
+            this.field = field;
+            this.comparator = comparator;
+            this.value = value;
+        }
+        
+        public Field getField() {
+            return field;
+        }
+        
+        public Comparator getComparator() {
+            return comparator;
+        }
+        
+        public String getValue() {
+            return value;
+        }
+
+        @Override
+        public final boolean equals(Object o) {
+            if (o instanceof Condition) {
+                Condition condition = (Condition) o;
+                return Objects.equals(field, condition.field)
+                    && Objects.equals(comparator, condition.comparator)
+                    && Objects.equals(value, condition.value);
+            }
+            return false;
+        }
 
-    public static Rule of(String id) {
-        return new Rule(id);
+        @Override
+        public final int hashCode() {
+            return Objects.hash(field, comparator, value);
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                .add("field", field)
+                .add("comparator", comparator)
+                .add("value", value)
+                .toString();
+        }
     }
 
-    public Rule(String id) {
-        Preconditions.checkNotNull(id);
 
+    public static class Action {
```

Comment:
@rouazana sad in chat:
> maybe if you s/action/set the intent will be more understandable

https://chat.linagora.com/openpaas/pl/f8kzuez3738uuy1u93c7397ccw

Reply:
Yes, but matthieu's suggestion seems better, something like Action { Set { mailboxIds } }

Reply:
I'll modify the POJO